woocommerce/woocommerce-android
# Show Progress when loading printing UI

Parent issue #4039

@kidinov I have requested a review from you since I touched the dimensions of the views in the payment flow. Let me know if you think the changes I made are safe. The content was jumping when I tried to show the progress without the changes.

Merge instructions
~1. Make sure #4254 is merged~
~2. Remove "do not merge" label~
3. Merge this PR

This PR displays a progress dialog when the user clicks on "Print receipt" as it usually takes 1-3seconds before the OS printing UI is shown. 

Note 1: I think we would ideally show a small progress directly on the button (something like [this](https://www.google.com/url?sa=i&url=https%3A%2F%2Fuxmovement.com%2Fbuttons%2Fwhen-you-need-to-show-a-buttons-loading-state%2F&psig=AOvVaw3jr9jylEnaOFOfgO115hU2&ust=1624658720917000&source=images&cd=vfe&ved=0CAoQjRxqFwoTCPDCpMqjsfECFQAAAAAdAAAAABAD)), but that's out of scope for the beta - I was thinking we could implement it later, wdyt @woocommerce/mobile-designers?

~Note 2: The cut-off send receipt button is a known issue not related to this PR.~

![Jun-24-2021 16-53-33](https://user-images.githubusercontent.com/2261188/123337390-bcdf3280-d50c-11eb-992f-a5220ac0d381.gif)

To test
1. Collect a payment - check if the dialog content doesn't jump from one step to another
2. Tap on "Print receipt" and notice a progress is shown
3. Notice the progress is not shown when you get back from the OS printing UI


Update release notes:
